ui/components/src/model/units/time.ts

```diff
@@ -73,13 +73,16 @@ export enum TimeIntlDuration {
   Years = 'year',
 }
 
-export function formatTime(value: number, unitOptions: TimeUnitOptions): string {
-  const decimalPlaces = unitOptions.decimal_places ?? DEFAULT_DECIMAL_PLACES;
-  const timeUnit: string = TimeIntlDuration[unitOptions.kind];
+export function formatTime(value: number, { kind, decimal_places }: TimeUnitOptions): string {
+  const decimalPlaces = decimal_places ?? DEFAULT_DECIMAL_PLACES;
+
+  const timeUnit: string = TimeIntlDuration[kind];
+  const isMonthOrYear = kind === 'Months' || kind === 'Years';
+
   const formatter = new Intl.NumberFormat('en-US', {
     style: 'unit',
     unit: timeUnit,
-    unitDisplay: 'long',
+    unitDisplay: isMonthOrYear ? 'long' : 'narrow',
```

Comment:
See tests to see the difference that this makes!

I made the distinction with `months` and `years` because:

* If I only used `narrow`, 300 minutes would display as `300m` and 300 months would display as `300m`, so these two units would be ambiguous. 
* If I'm going to spell out `months`, might as well spell out the unit that's bigger than `months` too. 
* I think months and years will be pretty rare, so it's not such a big impact if we spell them out. I think milliseconds and seconds will be way more common.

Reply:
Amazing, cool to see you make full use of the Intl.NumberFormat api!